src/kudu/consensus/raft_consensus.cc

```diff
@@ -4427,7 +4450,26 @@ void RaftConsensus::HandleProxyRequest(const ConsensusRequestPB* request,
 
 Status RaftConsensus::SetCompressionCodec(const std::string& codec) {
   LockGuard l(lock_);
-  return queue_->log_cache()->SetCompressionCodec(codec);
+  if (codec.empty()) {
+    LOG(INFO) << "Disabling compression";
+    codec_ = nullptr;
+    return Status::OK();
+  }
+
+  const CompressionCodec* comp_codec = nullptr;
+  auto codec_type = GetCompressionCodecType(codec);
+  if (codec_type != NO_COMPRESSION) {
+    auto status = GetCompressionCodec(codec_type, &comp_codec);
+    if (!status.ok()) {
+      LOG(ERROR) << "Failed to set compression codec";
+      return status;
+    }
+  }
+
+  LOG(INFO) << "Updating compression codec to: " << codec;
+  codec_ = const_cast<CompressionCodec*>(comp_codec);
```

Comment:
why was const_cast needed?

Reply:
`GetCompressionCodec()` takes `const CompressionCodec*` for some reason. So we get back a const which we need to cast to assign to codec_

Reply:
I think you can make codec_ const CompressionCodec* to prevent this

Reply:
const members can only be assigned in the ctor IIRC

Reply:
the ptr is not const. the object that the ptr is pointing to is const, so assigning the ptr should be fine.